build-farm/make-adopt-build-farm.sh

```diff
@@ -39,8 +39,9 @@ fi
 ## so needs to be special cased - on everthing else "uname" is valid
 if [ -z "$TARGET_OS" ]; then
   TARGET_OS=$(uname)
-  if [ "$OSTYPE" = "cygwin" ]; then TARGET_OS=windows ; fi
-  if [ "$OSTYPE" = "SunOS"  ]; then TARGET_OS=solaris ; fi
+  if [ "$OSTYPE"    = "cygwin" ]; then TARGET_OS=windows ; fi
+  if [ "$TARGET_OS" = "SunOS"  ]; then TARGET_OS=solaris ; fi
```

Comment:
This SunOs logic changed.  Was that deliberate?

Reply:
Yeah I considered doing that in a separate PR but (mostly becuase hardly anyone would care about Solaris) I put it in here. Happy to split it out again and get a free bonus PR out of it.

`$OSTYPE` won't be set to `SunOS` by default on Solaris so this needs to be using the output from `uname` - stored in `$TARGET_OS` - defined a few lines above to have the desired effect.

Reply:
(Also note I did mention that I'd changed solaris in the PR description)